File: regexlib/2021-5-15/javascript_test_file/regexlib_1141.js
/* 1141
 * ^[-+]?[0-9]+[.]?[0-9]*([eE][-+]?[0-9]+)?$
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:""+"0"*5000+"!1 _SLQ_2"
 */
var REGEX = /^[-+]?[0-9]+[.]?[0-9]*([eE][-+]?[0-9]+)?$/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + '0'.repeat(i*10000) + '!1 _SLQ_2'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}

File: regexlib/python_re_test_file/regexlib_1141.py
# 1141
# ^[-+]?[0-9]+[.]?[0-9]*([eE][-+]?[0-9]+)?$
# POLYNOMIAL
# nums:4
# POLYNOMIAL AttackString:""+"0"*10000+"!1 _SLQ_2"

import re
from time import perf_counter

regex = """^[-+]?[0-9]+[.]?[0-9]*([eE][-+]?[0-9]+)?$"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "0" * i * 10000 + "!1 _SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")